src/main/java/marquez/api/resources/DatasourceResource.java

```diff
@@ -91,10 +89,18 @@ public Response get(@PathParam("datasourceId") final DatasourceId datasourceId)
   @Produces(APPLICATION_JSON)
   public Response create(@NonNull final DatasourceRequest datasourceRequest)
       throws MarquezServiceException {
+
+    final ConnectionUrl connectionUrl;
+    try {
```

Comment:
The `try/catch` block can be removed when we add an exception mapper to handle thrown `IllegalArgumentException`s #403

Reply:
Although it may introduce changes originally not intended for the review, I'm happy to add it here with this PR.
Another option might be to open a new stacked PR atop #397.
LMK :)

Reply:
Will handle this at another time.